src/testshade/testshade.cpp

```diff
@@ -1239,7 +1239,24 @@ test_shade (int argc, const char *argv[])
 
     // We're done with the shading system now, destroy it
     shadergroup.reset ();  // Must release this before destroying shadingsys
+
+    TextureSystem *texturesys = shadingsys->texturesys();
```

Comment:
It seems a little sneaky to extract the texture system from the shadingsys, delete the latter and then print out info from the texture system.

I realize it works because this code is setup for the shared case, but is it maybe worth moving the `delete shadingsys` down a bit just in case? Or do you specifically need to catch errors that could have occurred during the destruction process?

Reply:
Exactly, I wanted to catch anything that happens during SS teardown.

But you are right. What I can do is ask for the TS up front, pass it to the SS ctr rather than implicitly relying on the shared one. Then I'll know that the TS outlives the SS.

Reply:
amended